### DISCUSSION TITLE
When I used promxy, I could get data for a short period of time, but I could not get data for more than 200 days, and no error message was reported

### DISCUSSION THREAD

**Discussion by SRE-Rick** [jacksontj/promxy#594] [Q&A]
When I used promxy, I could get data for a short period of time, but I could not get data for more than 200 days, and no error message was reported

**Comment by jacksontj** [comment]
If you are using VictoriaMetrics (and don't have caching disabled) this is a known issue (recently discussed in https://github.com/jacksontj/promxy/issues/537).

If not, I'd be curious about your setup as that shouldn't be the case (unless like the above VM bug; the downstream is not honoring the API contract)

**Reply by SRE-Rick** [reply]
![image](https://user-images.githubusercontent.com/86993048/236978579-688d1d2b-9598-4d6c-a632-652cb24c0d03.png)
Yes, my data source is VictoriaMetrics, I was able to query more than a year's history for one metric with caching disabled by the proxy, but when I used the grafana dashboard and showed several different monitors, the query was slow and I ended up not finding the data, What configurations should be changed to speed up queries and avoid being unable to retrieve data

**Reply by jacksontj** [reply]
Unfortunately there isn't generally a silver bullet for performance -- so to debug we'd need to dive into a lot more specifics. If you are still having this performance issue I'd recommend opening up an issue with (1) your configuration (2) trace logs of the scenarios and (3) ideally a repro case using public (or easy to replicate) data.
